modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/openapi_resource_impl.ftl

```diff
@@ -58,29 +58,35 @@ public class OpenAPIResourceImpl {
 	@GET
 	@Path("/openapi.{type:json|yaml}")
 	@Produces({MediaType.APPLICATION_JSON, "application/yaml"})
-	public Response getOpenAPI(@Context HttpServletRequest httpServletRequest, @PathParam("type") String type, @Context UriInfo uriInfo) throws Exception {
+	public Response getOpenAPI(@PathParam("type") String type) throws Exception {
 		Class<? extends OpenAPIResource> clazz = _openAPIResource.getClass();
 
 		try {
 			Method method = clazz.getMethod("getOpenAPI", HttpServletRequest.class, Set.class, String.class, UriInfo.class);
 
-			return (Response)method.invoke(_openAPIResource, httpServletRequest, _resourceClasses, type, uriInfo);
+			return (Response)method.invoke(_openAPIResource, _httpServletRequest, _resourceClasses, type, _uriInfo);
 		}
 		catch (NoSuchMethodException noSuchMethodException1) {
 			try {
 				Method method = clazz.getMethod("getOpenAPI", Set.class, String.class, UriInfo.class);
 
-				return (Response)method.invoke(_openAPIResource, _resourceClasses, type, uriInfo);
+				return (Response)method.invoke(_openAPIResource, _resourceClasses, type, _uriInfo);
 			}
 			catch (NoSuchMethodException noSuchMethodException2) {
 				return _openAPIResource.getOpenAPI(_resourceClasses, type);
 			}
 		}
 	}
 
+	@Context
+	private HttpServletRequest _httpServletRequest;
```

Comment:
I am pretty sure this change breaks the Global OpenAPI. We need to consider:

- If we leave the @Context annotation, the parameter could be set twice, once by JAX-RS and another one by Vulcan.
- Supposing from the previous point that the @Context annotation was removed, if an external Customer does not update the Vulcan version but uses REST Builder to create or update an existing API, his Vulcan version would not inject the `HttpServletRequest` properly

Reply:
apart from that, if the parameters are removed from the method signature, this invocation https://github.com/liferay/liferay-portal/blob/4c31c1c2f214ffd2cab84d598e0c9d179f73fb1b/modules/apps/headless/headless-discovery/headless-discovery-impl/src/main/java/com/liferay/headless/discovery/internal/jaxrs/resource/v1_0/HeadlessDiscoveryOpenAPIResourceImpl.java#L133 will fail always as the method will not be found

Reply:
I just stumbled upon this, what do you think if I keep both methods in rest builder?

Reply:
they are invoked from HeadlessDiscoveryOpenAPIResourceImpl, but if the method is invoked from JAX-RS, in some point the context needs to be set, either JAX-RS or Vulcan has to set it.

For example, if you open a browser and go to `http://localhost:8080/o/api`, then this method is invoked from JAX-RS: https://github.com/liferay/liferay-portal/blob/63e472506ed0f03248ca0301fb7ba0862a063954/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/OpenAPIResourceImpl.java#L62

Reply:
What if we keep the revert and update [this reflection ](https://github.com/liferay/liferay-portal/blob/master/modules/apps/headless/headless-discovery/headless-discovery-impl/src/main/java/com/liferay/headless/discovery/internal/jaxrs/resource/v1_0/HeadlessDiscoveryOpenAPIResourceImpl.java#L133) to check for both methods?

Reply:
Can we maybe keep both methods: 

- public Response getGlobalOpenAPI(@PathParam("type") String type) annotated with Jax-RS annotations
- public Response getOpenAPI(HttpServletRequest httpServletRequest, String type, UriInfo uriInfo) 
without annotations which will be used by HeadlessDiscoveryOpenAPIResourceImpl

Reply:
I'm pretty sure something like this could work for both of our cases, but I'm missing the broader picture of headless infrastructure. Can you please check if something like this would work https://github.com/liferay-headless/liferay-portal/pull/1017/commits/cdc631deef953b24de7dfb039ceeb73317e931be?